### PR TITLE
Add comparison of alevin-fry resolution modes

### LINKED ISSUE #104: Test use of cr-like-em resolution for alevin-fry
As we are narrowing in on the choice of configuration for alevin-fry, we have yet to test use of the `cr-like-em` resolution that is one of the resolutions recommended for use with the `splici` index and with USA mode. We should compare the `cr-like-em` method to `cr-like` alone, with the only difference being that `cr-like-em` resolves multi-mapped reads using the expectation maximization algorithm, while `cr-like` discards those reads. 

For this test, we will use 2 single-cell samples (SCPCR000126 and SCPCR000127) and 2 single-nuclei samples (SCPCR000220 and SCPCR000221). These samples will be run with the updated version of alevin-fry using `salign`, `splici`, and either `cr-like-em` or `cr-like`.

### PR DESCRIPTION
**Pull request by allyhawkins**
Closes #104. This PR tests the use of the `cr-like` resolution in comparison to the `cr-like-em` resolution mode in alevin-fry using similar per cell QC metrics discussed previously in benchmarking.

The prep for this notebook included updating docker containers for alevin-fry and salmon to use alevin-fry v0.4.0 and salmon 1.5.1. Two single cell (SCPCR000126 and SCPCR000127) and two single nuclei (SCPCR000220 and SCPCR000221) samples were run using `splici` with selective alignment, and either `cr-like` or `cr-like-em` with USA mode. The difference between the two modes is that the `cr-like-em` mode resolves multi-mapped reads using an expectation maximization algorithm, while `cr-like` throws them out. I am comparing these samples to the same samples run through cellranger. The dataframes used as input for this script were generated using the `benchmarking_generate_qc_df.R` script. 

From this analysis, `cr-like` looks as it did previously with similar numbers to cellranger when looking at shared cells and `cr-like-em` reporting slightly higher numbers. Unless there is a reason not to want to resolve multi-mapped reads, I think we could use `cr-like-em` moving forward.

### CODE REVIEW

**Review by jashapiro — APPROVED** [review]
This looks good, though as I said below, I think we can basically drop the "all cells" comparison, as I don't think it is particularly relevant to the questions we are after here.

I am curious about the genes that are being added here, but maybe that question will be addressed in #116. I'm going to go ahead and approve this, but probably look for a bit more detail in the gene comparisons.

File: analysis/quantifier-comparisons/07-alevin-fry-resolution-modes.Rmd [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+ggplot(cell_qc_common, aes(x = tool, y = detected, fill = alevin_resolution)) + 
+  geom_boxplot() + 
+  facet_wrap(~ sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Genes detected/cell") + 
+  xlab("") +
+  coord_cartesian(ylim = c(0,10000))
+```
+The `cr-like-em` resolution actually looks more similar to cellranger than the `cr-like` when looking at UMI/cell, but when looking at genes/cell you do see a shift upwards in the `cr-like-em`.
```
[the remark this answers is not in the document]
Comment (allyhawkins) [reply analysis/quantifier-comparisons/07-alevin-fry-resolution-modes.Rmd:201]: Based on the analysis in #116, I do think that the genes that are unique to `cr-like-em` are lower expressed genes, but I should do the direct comparison here. I have a feeling that use of the EM is increasing the number of lower covered genes that are resolved. I think I want to add a little bit more here to see where the additional information is coming from, as you suggest, to see if the EM is likely to be adding useful information or not.

[the remark this answers is not in the document]
Comment (allyhawkins) [reply analysis/quantifier-comparisons/07-alevin-fry-resolution-modes.Rmd:213]: I went ahead and looked at the genes that are shared between using `cr-like` and `cr-like-em` when looking at shared cells only and all of the genes overlap so it doesn't appear that one is detecting more genes than the other. What is interesting is that when you look at the distribution of the % of cells that genes are detected in between the two tools it is shifted slightly higher in the `cr-like-em` resolution. This appears to be only the case when you include lowly covered genes. If you filter for genes that are detected in > 5% of cells, the coverage appears uniform across both resolution methods. I added a short section to the notebook to show this. That makes me think we are probably going to be throwing those genes out and in the end the difference is negligible. If there are other comparisons or metrics that you think would be helpful in deciding which resolution to use let me know and I can add it in!

File: analysis/quantifier-comparisons/07-alevin-fry-resolution-modes.Rmd [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+glue::glue("Number of genes found in both {length(gene_overlap)}")
+```
+
+When you look at shared cells only and genes with expression > 0 and detected in at least 1 cell, `cr-like` and `cr-like-em`, 2,056 genes are not found to intersect.
+
+I'm going to look at gene expression and how this actually affects individual genes in another notebook. 
+
+```{r}
+sessioninfo::session_info()
+```
```
Comment (jashapiro) [inline analysis/quantifier-comparisons/07-alevin-fry-resolution-modes.Rmd:300]: something odd happened in the rendering here, not sure why.

File: analysis/quantifier-comparisons/07-alevin-fry-resolution-modes.Rmd [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    ~ purrr::map_df(.x, scpcaTools::rowdata_to_df, .id = "quant_dir"),
+    .id = "alevin_resolution"
+  )
+
+nucleus_rowdata_filtered <- nucleus_sces %>% 
+  purrr::map_df(
+    ~ purrr::map_df(.x, scpcaTools::rowdata_to_df, .id = "quant_dir"),
+    .id = "alevin_resolution"
+  )
+# merge into one dataframe
```
Comment (jashapiro) [inline analysis/quantifier-comparisons/07-alevin-fry-resolution-modes.Rmd:265]: This section is missing from the rendered notebook, somehow.

**Review by jashapiro — APPROVED** [review]
This looks good, but it seems like the notebook may need a rerender, as some parts seem to be missing from the html.

Answering one of your questions in the comments, I think it is fine to keep these notebooks separate!

**Pull request merged**